components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -532,7 +532,7 @@ public static String decode(String encodedStr) throws IdentityException {
             throw new IdentitySAML2ClientException("Error when decoding the SAML Request. " +
                     "Invalid arguments provided.", e);
         } catch (IOException e) {
-            throw IdentityException.error("Error when decoding the SAML Request.", e);
+            throw new IdentitySAML2ClientException("Error when decoding the SAML Request.", e);
```

Comment:
Are we catching those client exceptions in a single place? If the stack trace is printing in this client exception we need to handle that too.